Sources/CertLogic/CertLogic.swift

```diff
@@ -140,6 +143,36 @@ final public class CertLogicEngine {
     result = "{" + "\"\(Constants.external)\":" + "\(externalJsonString)" + "," + "\"\(Constants.payload)\":" + "\(payload)"  + "}"
     return result
   }
+
+    // Get List of Rules for Country by Code
+    private func getCustomListOfRulesForAll(filter: FilterParameter, issuerCountryCode: String) -> [Rule] {
+      var returnedRulesItems: [Rule] = []
+      var certTypeRulesWith = rules.filter { rule in
+        return rule.countryCode.lowercased() == filter.countryCode.lowercased()  && rule.certificateFullType == filter.certificationType && filter.validationClock >= rule.validFromDate && filter.validationClock <= rule.validToDate
+      }
+      if let region = filter.region {
+        certTypeRulesWith = certTypeRulesWith.filter { rule in
+          rule.region?.lowercased() == region.lowercased()
+        }
+      } else {
+        certTypeRulesWith = certTypeRulesWith.filter { rule in
+          rule.region == nil
+        }
+      }
+
+      let groupedCertTypeRulesWith = certTypeRulesWith.group(by: \.identifier)
+
+      groupedCertTypeRulesWith.keys.forEach { key in
+        let rules = groupedCertTypeRulesWith[key]
+        if let maxRules = rules?.max(by: { (ruleOne, ruleTwo) -> Bool in
+           return ruleOne.versionInt < ruleTwo.versionInt
+        }) {
+         returnedRulesItems.append( maxRules)
+        }
+      }
+
+      return returnedRulesItems
+    }
```

Comment:
Not really a good example for clean code. It's hard to grasp what you are doing here.

Reply:
its just an copy & paste of some other function to have the method similar to the original authors code

I think in this fork is more important to make minor changes